Packs/AWS-ACM/Integrations/AWS-ACM/AWS-ACM.py

```diff
@@ -326,35 +218,61 @@ def get_certificate(args):
-        list_certificates(demisto.args())
-    if demisto.command() == 'aws-acm-add-tags-to-certificate':
-        add_tags_to_certificate(demisto.args())
-    if demisto.command() == 'aws-acm-remove-tags-from-certificate':
-        remove_tags_from_certificate(demisto.args())
-    if demisto.command() == 'aws-acm-list-tags-for-certificate':
-        list_tags_for_certificate(demisto.args())
-    if demisto.command() == 'aws-acm-get-certificate':
-        get_certificate(demisto.args())
-except ResponseParserError as e:
-    return_error('Could not connect to the AWS endpoint. Please check that the region is valid.\n {error}'.format(
-        error=type(e)))
-    LOG(str(e))
-
-except Exception as e:
-    LOG(str(e))
-    return_error('Error has occurred in the AWS ACM Integration: {code}\n {message}'.format(
-        code=type(e), message=str(e)))
+def main():
+    try:
+        params = demisto.params()
+        aws_default_region = params.get('defaultRegion')
+        aws_role_arn = params.get('roleArn')
+        aws_role_session_name = params.get('roleSessionName')
+        aws_role_session_duration = params.get('sessionDuration')
+        aws_role_policy = None
+        aws_access_key_id = params.get('access_key')
+        aws_secret_access_key = params.get('secret_key')
+        verify_certificate = not params.get('insecure', True)
+        timeout = demisto.params().get('timeout')
+        retries = demisto.params().get('retries') or 5
+
+        validate_params(aws_default_region, aws_role_arn, aws_role_session_name, aws_access_key_id,
+                        aws_secret_access_key)
+
+        aws_client = AWSClient(aws_default_region, aws_role_arn, aws_role_session_name, aws_role_session_duration,
+                               aws_role_policy, aws_access_key_id, aws_secret_access_key, verify_certificate, timeout,
+                               retries)
+        args = demisto.args()
+        if demisto.command() == 'test-module':
+            test_function(aws_client)
+        if demisto.command() == 'aws-acm-describe-certificate':
+            describe_certificate(args, aws_client)
+        if demisto.command() == 'aws-acm-list-certificates':
+            list_certificates(args, aws_client)
+        if demisto.command() == 'aws-acm-add-tags-to-certificate':
+            add_tags_to_certificate(args, aws_client)
+        if demisto.command() == 'aws-acm-remove-tags-from-certificate':
+            remove_tags_from_certificate(args, aws_client)
+        if demisto.command() == 'aws-acm-list-tags-for-certificate':
+            list_tags_for_certificate(args, aws_client)
+        if demisto.command() == 'aws-acm-get-certificate':
+            get_certificate(args, aws_client)
+
+    except ResponseParserError as e:
+        return_error('Could not connect to the AWS endpoint. Please check that the region is valid.\n {error}'.format(
+            error=type(e)))
+        LOG(str(e))
```

Comment:
i think this is handeld already in the apimodule

Reply:
We don't check at the api module if the region is valid, we just require from the user to fill it, so I think we should leave it here